scalameta/trees/src/main/scala/scala/meta/Trees.scala

```diff
@@ -373,7 +373,14 @@ object Defn {
                    templ: Template) extends Defn with Member.Type {
     // TODO: hardcoded in the @ast macro, find out a better way
     // require(templ.stats.getOrElse(Nil).forall(!_.is[Ctor]))
-    require(ctor.mods.isEmpty && ctor.paramss.isEmpty)
+
+    // TODO this doesn't work because the Dialect in implicit
```

Comment:
We don't have a dialect at this point.

According to the spec it seems mods are allowed http://docs.scala-lang.org/sips/pending/trait-parameters.html